src/components/ChildHideCheckbox.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { useContext } from 'react';
+import { Checkbox, Tooltip } from 'antd';
+
+import { ChildCheckboxProps } from '../types';
+import { VisibilityContext } from '../AgentVisibilityContext';
+import { CheckboxState, tooltipMap } from '../constants';
+
+const ChildHideCheckbox: React.FunctionComponent<ChildCheckboxProps> = (
+  props: ChildCheckboxProps
+): JSX.Element => {
+  const { name, parentName } = props;
+  const { handleHideChildCheckboxChange, hiddenAgents } =
+    useContext(VisibilityContext);
+
+  const selections = hiddenAgents[parentName];
```

Comment:
```suggestion
// the selections are initialized with the parents names as keys, 
//so this will always exist, but typeScript doesn't know that. 
   const selections = hiddenAgents[parentName] || [];
```
(probably have to do some formatting on the comment to accept this change)

Reply:
ok ill do it manually and add it in the highlighted one as well